src/eq_schema/Group.js

```diff
@@ -1,15 +1,39 @@
+/* eslint-disable camelcase */
 const Block = require("./Block");
 const { getInnerHTML } = require("../utils/HTMLUtils");
+const { filter, isEmpty, compact } = require("lodash");
 
 class Group {
   constructor(id, title, pages, ctx) {
     this.id = `group${id}`;
     this.title = getInnerHTML(title);
-    this.blocks = this.buildBlocks(this.title, pages, ctx);
+    this.blocks = this.buildBlocks(this.title, pages, ctx, id);
+
+    if (!isEmpty(ctx.routingGotos)) {
+      this.filterContext(this.id, ctx);
+      const skipConditions = this.buildSkipConditions(this.id, ctx);
+
+      !isEmpty(skipConditions) ? (this.skip_conditions = skipConditions) : null;
+    }
+  }
+
+  filterContext(currentId, ctx) {
+    ctx.routingGotos = filter(
+      ctx.routingGotos,
+      rule => rule.group !== currentId
+    );
+  }
+
+  buildSkipConditions(currentId, ctx) {
+    return compact(
```

Comment:
Rather than removing nulls I wonder if it's better to map over a filtered list using `reject`

e.g. 

```js

reject(ctx.routingGotos, { group: currentId }).map(({when}) => ({when}));
```